### ISSUE TITLE
Getting permission errors at random while tagging succeeds

### ISSUE THREAD

**Issue by Gi11i4m** [jabbera/vsts-git-release-tag#23]
Just started using this extension in our builds. It's very neat, but randomly throws these errors:

```
2018-01-30T12:47:50.0239074Z git-tag-on-release-task: Version: 4.0.0
2018-01-30T12:47:51.4634774Z ##[error]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId={{redacted}}
2018-01-30T12:47:51.4637079Z ##[error]Unable to create ref: tags/2.0.2-22 UpdateStatus: 6 RepositoryId: {{redacted}} Commit: c274adce18a3645a304e1e971d8cab1c689555bc
2018-01-30T12:47:51.4642104Z ##[section]Finishing: Tag Development branch
```

This makes it seem as if our builds at random times only succeed partially, while tagging is still done and nothing actually went wrong. Any idea why this happens?

**Comment by jabbera** [comment]
2 questions. Are you doing this from a build or a release? I've never used it in a build so I'm not sure exactly how it would work.

Second: Are you re-using the same tag but against a different commit? This needs force permission. This can be confirmed by setting system.debug=true and posting the log from the task here.

**Comment by Gi11i4m** [comment]
We're doing this from release. Everytime we deploy to TEST the Development branch gets tagged with a unique version number, so the same tag should never be reused (unless we redeploy to TEST, which almost doesn't happen).

Where would I set system.debug = true?

**Comment by jabbera** [comment]
Hi

To enabled debugging, create a release variable called system.debug and set it's value to true.

**Comment by Gi11i4m** [comment]
The logs revealed what the problem is, and now I don't understand why a lot of times it does succeed.

We have two builds, one for the backend and one for the frontend. This is in the same repository, but every time backend code is committed the backend build starts and same for frontend.

Apparently this extension tries to tag the branch for every artifact. This makes sense because they come from different commits, but it's not the behavior we want. It should tag the latest commit from these two artifacts. I guess that's not really possible.

Is there anything we can do to avoid the error?

**Comment by jabbera** [comment]
Could it be succeeding when a single commit includes both front and backend changes, meaning both builds are at the same commit. We handle this situation correctly already by detecting an existing commit with the same ref id and not doing the tag. 

I'll look into ways to avoid this.

**Comment by jabbera** [comment]
Hey, give this the good old college try: https://github.com/jabbera/vsts-git-release-tag/releases/tag/5.0

I'm not going to put it out to the public yet, but let me know how it goes.

**Comment by Gi11i4m** [comment]
Awesome! I'll try it out asap, thanks!

**Comment by Gi11i4m** [comment]
Also to answer your previous question, normally it never happens that one commit includes both builds. The backend is built when backend code is checked in (trigger is on a certain folder) and same for frontend.

**Comment by jabbera** [comment]
published as 5.0.1

**Comment by Gi11i4m** [comment]
I couldn't test this out since we're using VSTS and not TFS. However, we're using your latest version now and for the past 6 releases it's always succeeded, so seems good!
